### PR TITLE
Fixed failing Event Log tests.

### PR DESCRIPTION
**Pull request by harshjain2**

### CODE REVIEW

**Comment by codito** [comment]
Suggest to run `test.cmd` and `test.cmd -p acceptance -filter EventLogCollectorTests` 100 times each to validate the fix.

**Comment by harshjain2** [comment]
@dotnet_bot test this please.

**Comment by harshjain2** [comment]
@dotnet-bot test this please.

**Comment by harshjain2** [comment]
@dotnet-bot test this please.

File: src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionTestCaseEventSender.cs [hunk 1]
```diff
@@ -90,6 +90,12 @@ public void Close()
         public void SendTestCaseStart(TestCaseStartEventArgs e)
         {
             this.communicationManager.SendMessage(MessageType.DataCollectionTestStart, e);
+
+            var message = this.communicationManager.ReceiveMessage();
```
Comment (acesiddhu) [inline src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionTestCaseEventSender.cs]: >this.communicationManager.ReceiveMessage(); [](start = 26, length = 43)

if this call not sync? why do we need a while loop. for eg in SendTestCaseEnd below i dont see us doing the while loop thing

Comment (harshjain2) [reply src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionTestCaseEventSender.cs]: This PR is not fully baked yet. 
I pushed this to verify if Event Log DataCollector Acceptance tests are working or not.

File: test/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector.UnitTests/EventLogContainerTests.cs [hunk 2]
```diff
@@ -58,9 +58,8 @@ public EventLogContainerTests()
                 this.dataCollectionContext);
         }
 
-        [Ignore]
         [TestMethod]
-        // Ignoring because this tests is failing consistently in PR build.
+        [Ignore]
```
Comment (codito) [inline test/DataCollectors/Microsoft.TestPlatform.Extensions.EventLogCollector.UnitTests/EventLogContainerTests.cs]: Shouldn't the product fix enable this test?

**Review by codito — APPROVED** [review]

**Pull request merged**